docs/docsite/rst/network/dev_guide/developing_resource_modules_network.rst

```diff
@@ -0,0 +1,379 @@
+        │   ├── myos_facts.py
+        │   └── myos_interfaces.py
+        ├── LICENSE.txt
+        ├── module_utils
+        │   ├── __init__.py
+        │   └── network
+        │       ├── __init__.py
+        │       └── myos
+        │           ├── argspec
+        │           │   ├── facts
+        │           │   │   ├── facts.py
+        │           │   │   └── __init__.py
+        │           │   ├── __init__.py
+        │           │   └── interfaces
+        │           │       ├── __init__.py
+        │           │       └── interfaces.py
+        │           ├── config
+        │           │   ├── base.py
+        │           │   ├── __init__.py
+        │           │   └── interfaces
+        │           │       ├── __init__.py
+        │           │       └── interfaces.py
+        │           ├── facts
+        │           │   ├── base.py
+        │           │   ├── facts.py
+        │           │   ├── __init__.py
+        │           │   └── interfaces
+        │           │       ├── __init__.py
+        │           │       └── interfaces.py
+        │           ├── __init__.py
+        │           └── utils
+        │               ├── __init__.py
+        │               └── utils.py
+        └── README.md
+
+
+Using the collection
+--------------------
+
+This example shows how to use the generated collection in a playbook:
+
+ .. code-block:: yaml
+
+     ----
+     - hosts: myos101
+       gather_facts: False
+       tasks:
+       - cidrblock.my_collection.myos_interfaces:
+         register: result
+       - debug:
+           var: result
+       - cidrblock.my_collection.myos_facts:
+       - debug:
+           var: ansible_network_resources
+
+
+Using the role
+--------------
+
+This example shows how to use the generated role in a playbook:
+
+.. code-block:: yaml
+
+    - hosts: myos101
+      gather_facts: False
+      roles:
+      - my_role
+
+    - hosts: myos101
+      gather_facts: False
+      tasks:
+      - myos_interfaces:
+        register: result
+      - debug:
+          var: result
+      - myos_facts:
+      - debug:
+          var: ansible_network_resources
+
+
+Resource module structure and workflow
+======================================
+
+The resource module structure includes the following components:
+
+  Module
+    ``library/<ansible_network_os>_<resource>.py``.
+    Imports the ``module_utils`` resource package and calls ``execute_module`` API
+
+    .. code-block:: python
+
+      def main():
+          result = <resource_package>(module).execute_module()
+
+  Module argspec
+    ``module_utils/<ansible_network_os>/argspec/<resource>/``.
+    Argspec for the resource.
+
+  Facts
+    ``module_utils/<ansible_network_os>/facts/<resource>/``.
+    - Populate facts for the resource.
+    - Entry in ``module_utils/<ansible_network_os>/facts/facts.py`` for ``get_facts`` API to keep ``<ansible_network_os>_facts`` module and facts gathered for the resource module in sync for every subset.
```

Comment:
`Make entry of Resource subset in FACTS_RESOURCE_SUBSETS list in module_utils/<ansible_network_os>/facts/facts.py` to make facts collection working.